crypto/key.go

```diff
@@ -251,6 +251,18 @@ func (key *Key) GetPublicKey() (b []byte, err error) {
 
 // --- Key object properties
 
+// CanSign returns true if any of the subkeys can be used for signing.
+func (key *Key) CanSign() bool {
+	_, canSign := key.entity.SigningKey(getNow())
+	return canSign
```

Comment:
IMHO we've got a naming problem here: you would expect a key that returns true on `Key.CanSign()` to be able to sign. But a public key might pass this test too, since we are just checking if there is a valid signing entity, without checking if this has the private key material.

I'd recommend renaming this function so that we can still check the entity without causing confusion, and return true on public keys.

```suggestion
// CanVerify returns true if any of the subkeys can be used for verification.
func (key *Key) CanVerify() bool {
	_, canVerify := key.entity.SigningKey(getNow())
	return canVerify
```